tests/testthat/test-allow_partial_matching.R

```diff
@@ -0,0 +1,251 @@
+                             grader_args = list(
+                               user_quo = user_quo, 
+                               solution_quo = solution_quo
+                             )
+  )
+
+  
+  
+  
+  expect_false(pmatch_FALSE$correct)
+  expect_match(object = pmatch_FALSE$message,regexp = "p = \"no\"")
+  expect_match(object = pmatch_FALSE$message,regexp = "p1 = \"no\"")
+  
+  
+})
+
+test_that("allow_partial_matching works errors multi arg type", {
+  
+  ff <- function(chr = "yes", fun = ls, call =ls(), bool = TRUE){
+    print("youpi")
+  }
+  
+  solution_quo <- as.expression(quote(ff(chr = "yes", fun = ls, call =ls(), bool = TRUE)))
+  user_quo <- as.expression(quote(ff(ch = "yes", fu = ls, cal =ls(), boo = TRUE)))
+  
+  
+  pmatch_FALSE <- grade_code(allow_partial_matching = FALSE,
+                             grader_args = list(
+                               user_quo = user_quo, 
+                               solution_quo = solution_quo
+                             )
+  )
+
+  
+  
+  
+  expect_false(pmatch_FALSE$correct)
+  expect_match(object = pmatch_FALSE$message,regexp = "ch = \"yes\"")
+  expect_match(object = pmatch_FALSE$message,regexp = "chr = \"yes\"")
+  expect_match(object = pmatch_FALSE$message,regexp = "boo = TRUE")
+  expect_match(object = pmatch_FALSE$message,regexp = "bool = TRUE")
+  expect_match(object = pmatch_FALSE$message,regexp = "cal = ls()")
+  expect_match(object = pmatch_FALSE$message,regexp = "call = ls()")
+  expect_match(object = pmatch_FALSE$message,regexp = "fun = ls")
+  expect_match(object = pmatch_FALSE$message,regexp = "fu = ls")
+  
+  
+})
+
+test_that("allow_partial_matching works with multiple matches", {
+  
+  ff <-  function(ab, abc, abcd) { 1 }
+  
+  
+  solution_quo <- as.expression(quote(ff(ab = 1)))
+  user_quo <- as.expression(quote(ff(ab = 1)))
+  pmatch_FALSE <- grade_code(allow_partial_matching = FALSE,
+                             grader_args = list(
+                               user_quo = user_quo, 
+                               solution_quo = solution_quo
+                             ))
+  
+  expect_true(pmatch_FALSE$correct)
+  
+  
+  solution_quo <- as.expression(quote(ff(a = 1)))
+  user_quo <- as.expression(quote(ff(a = 1)))
+  pmatch_FALSE <- grade_code(allow_partial_matching = FALSE,
+                             grader_args = list(
+                               user_quo = user_quo, 
+                               solution_quo = solution_quo
+                             ))
+  
+  expect_true(pmatch_FALSE$correct)
+  
+  
+  
+  solution_quo <- as.expression(quote(ff(abc=3)))
+  user_quo <- as.expression(quote(ff(ab = 1,abc = 1)))
+  
+  set.seed(1)
+  pmatch_FALSE <- grade_code(allow_partial_matching = FALSE,
+                             grader_args = list(
+                               user_quo = user_quo, 
+                               solution_quo = solution_quo
+                             ))
+  set.seed(1)
```

Comment:
What are the `set.seed()` calls for?

```suggestion
  pmatch_FALSE <- grade_code(allow_partial_matching = FALSE,
                             grader_args = list(
                               user_quo = user_quo, 
                               solution_quo = solution_quo
                             ))
```

Reply:
to be sure that the message contain the same praise phrase :)